samza-yarn/src/main/scala/org/apache/samza/webapp/ApplicationMasterRestServlet.scala

```diff
@@ -19,44 +19,41 @@
 
 package org.apache.samza.webapp
 
+import java.{lang, util}
+
 import org.apache.samza.clustermanager.SamzaApplicationState
 import org.scalatra._
 import scalate.ScalateSupport
 import org.apache.samza.config.Config
-import org.apache.samza.job.yarn.{YarnAppState, ClientHelper}
+import org.apache.samza.job.yarn.{ClientHelper, YarnAppState}
 import org.apache.samza.metrics._
+
 import scala.collection.JavaConverters._
 import org.apache.hadoop.yarn.conf.YarnConfiguration
 import java.util.HashMap
-import org.apache.samza.serializers.model.SamzaObjectMapper
 
-class ApplicationMasterRestServlet(samzaConfig: Config, samzaAppState: SamzaApplicationState, state: YarnAppState, registry: ReadableMetricsRegistry) extends ScalatraServlet with ScalateSupport {
-  val yarnConfig = new YarnConfiguration
-  val client = new ClientHelper(yarnConfig)
-  val jsonMapper = SamzaObjectMapper.getObjectMapper
-
-  before() {
-    contentType = "application/json"
-  }
+import org.apache.samza.serializers.model.SamzaObjectMapper
+import org.codehaus.jackson.map.ObjectMapper
 
-  get("/metrics") {
-    val metricMap = new HashMap[String, java.util.Map[String, Object]]
+object ApplicationMasterRestServlet {
+  def getMetrics(jsonMapper: ObjectMapper, metricsRegistry: ReadableMetricsRegistry) = {
+    val metricMap = new HashMap[String, util.Map[String, Object]]
 
     // build metric map
-    registry.getGroups.asScala.foreach(group => {
+    metricsRegistry.getGroups.asScala.foreach(group => {
       val groupMap = new HashMap[String, Object]
 
-      registry.getGroup(group).asScala.foreach {
+      metricsRegistry.getGroup(group).asScala.foreach {
         case (name, metric) =>
           metric.visit(new MetricsVisitor() {
             def counter(counter: Counter) =
-              groupMap.put(counter.getName, counter.getCount: java.lang.Long)
+              groupMap.put(counter.getName, counter.getCount: lang.Long)
```

Comment:
Should this be .asInstanceOf[Long]? Also, are the annotations required?

Reply:
My scala foo is weak, but after looking into it. It looks like ascription does a better job of disambiguation than casting. I'd rather not trade compile-time errors for runtime-errors just for the sake of convention. Is there something more you know about this behavior that I'm missing?

Reply:
No, just asking why they're inconsistent.

Looks like the annotation is required since the original return type is `long` which gets converted to scala.Long which does not extend Object. Type annotation makes it box to a java.util.Long. So makes sense as is.